pims/norpix_reader.py

```diff
@@ -150,12 +183,17 @@ def _verify_frame_no(self, i):
         if i >= self._image_count or i < 0:
             raise ValueError("Frame number is out of range: " + str(i))
 
+    def set_process_func(self, process_func):
```

Comment:
This was an attempt to use the mechanisms that already exist in PIMS.

Since the process_func can be passed to __init__, but you generally need to read the header before you know which process_func to use, the alternative was to open the file, get the header information, close the file, and re-open it with the appropriate process_func.